editor/script_editor_debugger.cpp

```diff
@@ -342,6 +342,55 @@ void ScriptEditorDebugger::_scene_tree_request() {
 	ppeer->put_var(msg);
 }
 
+/// Populates inspect_scene_tree recursively given data in nodes.
+/// Returns the number of items parsed in nodes from current_index.
+///
+/// Given a nodes array like [R,A,B,C,D,E] the following Tree
+/// will be generated, assuming filter is an empty String.
+///
+/// R
+/// |-A
+/// | |-B
+/// | | |-C
+/// | |
+/// | |-D
+/// |
+/// |-E
+///
+int ScriptEditorDebugger::_update_scene_tree(TreeItem *parent, const Array &nodes, int current_index) {
+	String filter = EditorNode::get_singleton()->get_scene_tree_dock()->get_filter();
+	String item_text = nodes[current_index + 1];
+	bool keep = filter.is_subsequence_ofi(item_text);
+
+	TreeItem *item = inspect_scene_tree->create_item(parent);
+	item->set_text(0, item_text);
+	ObjectID id = ObjectID(nodes[current_index + 3]);
+	Ref<Texture> icon = EditorNode::get_singleton()->get_class_icon(nodes[current_index + 2], "");
+	if (icon.is_valid()) {
+		item->set_icon(0, icon);
+	}
+	item->set_metadata(0, id);
+
+	int children_count = nodes[current_index];
+	// Tracks the number of items parsed in nodes
+	int items_count = 1;
```

Comment:
why not just a bool? return a bool if items exist and default it to false..

Reply:
Looks like I misunderstood the code, the added comments made it more explicit.